internal/handler/userHandler_test.go

```diff
@@ -0,0 +1,290 @@
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code)
+	assert.Equal(t, "Error while binding JSON", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.NotEmpty(t, responseBody.Field)
+
+	fields, _ := convertToMap(responseBody.Field)
+	assert.Equal(t, "required", fields["firstName"])
+	assert.Equal(t, "required", fields["lastName"])
+	assert.Equal(t, "required", fields["email"])
+	assert.Equal(t, "age must be greater than 18", fields["age"])
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+
+func TestBindAgeAndEmailJsonFields(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "John",
+		LastName:  "Doe",
+		Email:     "invalid-email",
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code)
+	assert.Equal(t, "Error while binding JSON", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.NotEmpty(t, responseBody.Field)
+
+	fields, _ := convertToMap(responseBody.Field)
+	assert.Equal(t, "not a valid email address", fields["email"])
+	assert.Equal(t, "required", fields["age"])
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+
+func TestFindById(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+	id := uuid.New().String()
+
+	pathParam := []gin.Param{
+		{
+			Key:   "id",
+			Value: id,
+		},
+	}
+	ctx.Params = pathParam
+
+	user, _ := models.NewUser(id, "John", "Doe", "johndoe@email.com", 18)
+	mockService.EXPECT().GetUserById(ctx, id).Return(&user, nil).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.FindById(ctx)
+	var responseBody response.UserResponseDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusOK, recorder.Code)
+	assert.Equal(t, user.GetID(), responseBody.ID)
+	assert.Equal(t, user.GetFirstName(), responseBody.FirstName)
+	assert.Equal(t, user.GetLastName(), responseBody.LastName)
+	assert.Equal(t, user.GetEmail(), responseBody.Email)
+	assert.Equal(t, user.GetAge(), responseBody.Age)
+	mockService.AssertExpectations(t)
```

Comment:
These lines are not required, see comment in TestSaveUserSuccess.

Reply:
Fixed